api-tests/specs/practitioner-service/linkperson.js

```diff
@@ -0,0 +1,125 @@
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse1.status).to.equal(HttpStatus.OK)
+
+    });
+
+    it('Given a Person exists who is not linked to a User When I submit an API request to link the Person to a User who is already linked to a Person in this trial Then the Person and User are NOT linked and I receive an error notification ', async () => {
+
+        //post a request to practitioner end point to create second Practitioner
+        const headers = await utils.getHeadersWithAuth()
+        let fetchResponse = await fetch(conf.baseUrl + practitionerEndpointUri, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(requests.createSecondPerson),
+        })
+        let response2 = await fetchResponse.json()
+        secondPersonId = response2.id;
+
+        //link person to a user who is already linked
+        let linkUserJSON = requests.linkUser;
+        linkpersonEndpointUri2 = conf.baseUrl + linkpersonEndpointUri.replace("{personId}", firstPersonId);
+        
+        //check application response confirms link failure
+        
+        let fetchResponse2 = await fetch(linkpersonEndpointUri2, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse2.status).to.equal(HttpStatus.UNPROCESSABLE_ENTITY)
+
+    });
+
+    it('When I submit an API request with missing Practitioner ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
+
+        //post a request to practitioner end point to create third Practitioner
+        const headers = await utils.getHeadersWithAuth()
+        let fetchResponse = await fetch(conf.baseUrl + practitionerEndpointUri, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(requests.createSecondPerson),
+        })
+        let response3 = await fetchResponse.json()
+        thirdPersonId = response3.id;
+                
+        //link person to a user with request missing practitioner ID
+        let linkUserJSON = requests.linkUser;
+        linkpersonEndpointUri3 = conf.baseUrl + linkpersonEndpointUri.replace("{personId}", " ");
+
+        //check application response confirms link failure
+        
+        let fetchResponse3 = await fetch(linkpersonEndpointUri3, {
+            headers: headers,
+            method: 'POST',
+            body: JSON.stringify(linkUserJSON),
+        })
+        expect(fetchResponse3.status).to.equal(HttpStatus.UNPROCESSABLE_ENTITY)
+
+    });
+
+    it('When I submit an API request with missing User ID to link a Person who is NOT linked to a User to a User who is not linked to a Person in this trial Then the Person and User are NOT linked and I receive a failure acknowledgement ', async () => {
```

Comment:
Given a Person who is NOT linked to a User to a User who is not linked to a Person in this trial 
When I submit an API request with missing User ID to link the Person and User 
Then the Person and User are NOT linked and I receive a failure
?

Reply:
Copied directly from the ARTS-185 story, the original link person story